src/transformers/models/voxtral/configuration_voxtral.py

```diff
@@ -0,0 +1,254 @@
+        text_config (`Union[AutoConfig, dict]`, *optional*, defaults to `LlamaConfig`):
+            The config object or dictionary of the text backbone.
+        audio_token_index (`int`, *optional*, defaults to 151646):
+            The image token index to encode the image prompt.
+
+    Example:
+
+    ```python
+    >>> from transformers import VoxtralForConditionalGeneration, VoxtralConfig, VoxtralEncoderConfig, Qwen2Config
+
+    >>> # Initializing a VoxtralEncoder config
+    >>> audio_config = VoxtralEncoderConfig()
+
+    >>> # Initializing a Qwen2 config
+    >>> text_config = Qwen2Config()
+
+    >>> # Initializing a Voxtral configuration
+    >>> configuration = VoxtralConfig(audio_config, text_config)
+
+    >>> # Initializing a model from the qwen2-audio style configuration
+    >>> model = VoxtralForConditionalGeneration(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "voxtral"
+    sub_configs = {"text_config": AutoConfig, "audio_config": AutoConfig}
+
+    _default_text_config_kwargs = {
+        "vocab_size": 131072,
+        "max_position_embeddings": 131072,
+        "hidden_size": 3072,
+        "intermediate_size": 8192,
+        "num_hidden_layers": 30,
+        "num_attention_heads": 32,
+        "num_key_value_heads": 8,
+        "hidden_act": "silu",
+        "initializer_range": 0.02,
+        "rms_norm_eps": 1e-05,
+        "pretraining_tp": 1,
+        "use_cache": True,
+        "rope_theta": 100000000.0,
+        "rope_scaling": None,
+        "attention_bias": False,
+        "attention_dropout": 0.0,
+        "mlp_bias": False,
+        "head_dim": 128,
+        "tie_word_embeddings": False,
+        "bos_token_id": 1,
+        "eos_token_id": 2,
+        "sliding_window": None,
+    }
+
+    def __init__(
+        self,
+        audio_config=None,
+        text_config=None,
+        audio_token_id=24,
+        projector_hidden_act="gelu",
+        **kwargs,
+    ):
+        self.audio_token_id = audio_token_id
+        self.projector_hidden_act = projector_hidden_act
+
+        if isinstance(audio_config, dict):
+            audio_config["model_type"] = (
+                audio_config["model_type"] if "model_type" in audio_config else "voxtral_encoder"
+            )
+            audio_config = CONFIG_MAPPING[audio_config["model_type"]](**audio_config)
+        elif audio_config is None:
+            audio_config = CONFIG_MAPPING["voxtral_encoder"]()
+
+        self.audio_config = audio_config
+
+        if isinstance(text_config, dict):
+            text_config["model_type"] = text_config["model_type"] if "model_type" in text_config else "llama"
+            text_config = CONFIG_MAPPING[text_config["model_type"]](
+                **{**self._default_text_config_kwargs, **text_config}
+            )
+        elif text_config is None:
+            text_config = CONFIG_MAPPING["llama"](**self._default_text_config_kwargs)
```

Comment:
same here I'm not sure we'd want to allow defaults here

Reply:
happy to remove and see how it goes!